### PR TITLE
feat(jest): add jest unit testing support to `frontend`

### PR DESCRIPTION
**Pull request by stephenliang**
This change adds jest and react-testing-library to allow for unit testing. The stub samples were updated to include copilot generated unit tests, these are meant to be removed when actual components are created.

The following tools were added:

* jest: unit testing framework
* @swc/jest: transpile typescript jest code to js
* jsdom: emulate the dom in unit tests

### CODE REVIEW

File: frontend/packages/dsco/src/stub-section/__tests__/StubSection.test.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * These tests were generated via Copilot for scaffolding and should be removed when actual components are implemented.
```
Comment (kelbyhawn) [inline frontend/packages/dsco/src/stub-section/__tests__/StubSection.test.tsx:2]: Is a followup ticket needed for this?

**Review by kelbyhawn — APPROVED** [review]
One minor question, but someone else with more Jest experience should take a look too.

File: frontend/packages/dsco/src/stub-section/__tests__/StubSection.test.tsx (hunk 1 above)
Comment (stephenliang) [reply frontend/packages/dsco/src/stub-section/__tests__/StubSection.test.tsx:2]: Created: https://codedotorg.atlassian.net/browse/ACQ-2844

**Review by bethanyaconnor — APPROVED** [review]
I also don't know jest very well but looks reasonable to me.

**Pull request merged**